### PR TITLE
Channel message dto

### PR DESCRIPTION
**Pull request by Isioblast**
ChannelMessage dto are added and test file is updated, every tests pass except GameStat test. This is for another issue.

### CODE REVIEW

File: backend/src/channel-message/dto/update-channel-message.input.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Field, InputType } from '@nestjs/graphql'
+import { IsNotEmpty, IsString, MaxLength } from 'class-validator'
+
+@InputType()
+export class UpdateChannelMessageInput {
```
Comment (Chicanoss) [inline backend/src/channel-message/dto/update-channel-message.input.ts:5]: We should use 'extends PartialType' no ?

**Review by Chicanoss — CHANGES_REQUESTED** [review]

File: backend/src/channel-message/channel-message.service.ts [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
 @Injectable()
 export class ChannelMessageService {
   constructor(private prisma: PrismaService) {}
 
   async create(
-    data: Prisma.ChannelMessageCreateInput
+    createChannelMessageInput: CreateChannelMessageInput
   ): Promise<ChannelMessage> {
     return this.prisma.channelMessage.create({
```
Comment (kbarbry) [inline backend/src/channel-message/channel-message.service.ts:14]: replace createChannelMessageInput by data, we keep the same norm
if return this.prisma.channelMessage.create({ data }) doesn't work then there is a problem on your dto, they must be compatible without re-setting every field on your data

File: backend/src/channel-message/channel-message.service.ts [hunk 3]
```diff
@@ -65,28 +65,15 @@ export class ChannelMessageService {
...
-    if (data.user) {
-      throw new ChannelMessageExceptionTryingToUpdateSenderID()
-    }
-    if (!data.updatedAt) {
-      data.updatedAt = new Date()
-    }
     return this.prisma.channelMessage.update({
       where: {
         id
       },
```
Comment (kbarbry) [inline backend/src/channel-message/channel-message.service.ts:73]: same here

File: backend/src/channel-message/dto/update-channel-message.input.ts (hunk 1 above)
Comment (kbarbry) [reply backend/src/channel-message/dto/update-channel-message.input.ts:5]: yes

**Review by Chicanoss — APPROVED** [review]

File: backend/src/channel-message/dto/create-channel-message.input.ts [hunk 4]
```diff
@@ -1,24 +1,17 @@
...
-  MaxLength
-} from 'class-validator'
+import { IsString, IsUUID, Length, MaxLength } from 'class-validator'
 
 @InputType()
 export class CreateChannelMessageInput {
   @Field(() => String)
-  @IsString({ always: true })
-  @IsNotEmpty()
+  @IsString()
```
Comment (kbarbry) [inline backend/src/channel-message/dto/create-channel-message.input.ts:7]: edit MaxLength by Length(1, 2000)

**Review by kbarbry — CHANGES_REQUESTED** [review]

**Review by kbarbry — APPROVED** [review]

**Pull request merged**